network.go

```diff
@@ -272,6 +272,124 @@ func getInterface(netHandle *netlink.Handle, link netlink.Link) *pb.Interface {
 ////////////
 // Routes //
 ////////////
+func (s *sandbox) updateRoutes(netHandle *netlink.Handle, requestedRoutes *pb.Routes) (resultingRoutes *pb.Routes, err error) {
+	//	s.network.routesLock.Lock()
+	//	defer s.network.routesLock.Unlock()
+	if netHandle == nil {
+		netHandle, err = netlink.NewHandle()
+		if err != nil {
+			return nil, err
+		}
+		defer netHandle.Delete()
+	}
+
+	//
+	// First things first, let's blow away all the existing routes.  The updateRoutes function
+	// is designed to be declarative, so we will attempt to create state matching what is
+	// requested, and in the event that we fail to do so, will return the error and final state.
+	//
```

Comment:
The same issue:

I think it's also better to calculate differences with current route rules, and find routes to-delete/to-add, then only apply the adding and deleting for routes in smaller slices.

This can make sure we lost less if any error happens. E.g. after all routes are cleaned, we fail to add any route, so none route will exist.

Reply:
@WeiZhang555 -- but we don't expect this to happen, and we are in a bad state if it does, anyway, right?

Reply:
@egernst Yes, you're right. It will be in bad state if any error happens, and runtime may need to redo everything whatever.

I don't have a strong opinion on how to handle this, this part lgtm.